scipy/stats/_continuous_distns.py

```diff
@@ -4659,6 +4659,38 @@ def _entropy(self):
         # https://en.wikipedia.org/wiki/Logistic_distribution
         return 2.0
 
+    def fit(self, data, *args, **kwds):
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
+
+        # if user has provided `floc` or `fscale`, fall back on super fit
+        # method. This scenario is not suitable for solving a system of
+        # equations
+        if floc is not None or fscale is not None:
+            return super(logistic_gen, self).fit(data, *args, **kwds)
+
+        # rv_continuous provided guesses
+        loc, scale = self._fitstart(data)
+        # account for user provided guesses
+        loc = kwds.pop('loc', loc)
+        scale = kwds.pop('scale', scale)
+
+        # the maximum likelihood estimators `a` and `b` of the location and
+        # scale parameters are roots of the two equations described in `func`.
+        # Source: Statistical Distributions, 3rd Edition. Evans, Hastings, and
+        # Peacock (2000), Page 130
+        def func(input, data):
+            a, b = input
+            n = len(data)
+            x1 = np.sum(np.exp((data - a) / b) /
```

Comment:
Repeating comment from https://github.com/scipy/scipy/pull/12737#discussion_r476308071

It seems like precalculating `norm(?) = (data - a) / b`  and `np.exp(norm)` is could save a fair bit of work here.

```
norm = (data - a) / b  # idk what the name should be
exp_norm = np.exp(norm)

x1 = np.sum(norm / (1 + exp_norm)) - n/2
x2 = np.sum(norm * (norm - 1) / (norm + 1)) - n
```

Reply:
I agree with @sethtroisi's suggestion, those expressions should be computed once and reused.

Reply:
Added